### PR TITLE
[PA-227] Remotecamp Google Sheet To Jira Tool

### PR DESCRIPTION
**Pull request by umutozk**
Jira link: https://jira.devfactory.com/browse/PA-227

#### Changes
- Initial implementation for RemoteCamp Google sheet to Jira ticket tool.

#### Notes
- Since they don't want to share actual sheets with real data and their Jira project is not visible to our service account, limited testing was done with a copy of sample sheet.

### CODE REVIEW

**Comment by unlocked** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

**Comment by unlocked** [comment]
Hi, I'm Max Tolstykh from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely would you recommend the Code Review service to another developer?</b>
    
Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/1&entry.1163746017=10) \] Very likely
    
> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

File: src/main/java/com/devfactory/processautomation/service/process/TicketServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private static final int SCORE_INDEX = 2;
+    private static final int DESCRIPTION_INDEX = 3;
+    private static final int LINK_TO_IQB_INDEX = 4;
+    private static final String MODULE_FIELD_ID = "customfield_12861";
+    private static final String SCORE_FIELD_ID = "customfield_12863";
+    private static final String LINK_TO_IQB_FIELD_ID = "customfield_12864";
+    private static final String FIELD_SUMMARY = "summary";
+
+    @Override
+    public JiraTicket createTicket(final List<CellValue> cellValues) {
```
Comment (josemartinsquest) [inline src/main/java/com/devfactory/processautomation/service/process/TicketServiceImpl.java]: If this are all non mandatory fields, we should pass this to the JiraTicket it self..

No need to do this in every single project.

if (!StringUtils.isBlank(value)) {
...

File: src/main/java/com/devfactory/processautomation/service/process/TicketServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        addSimpleCustomField(ticket, SCORE_FIELD_ID, getCellValue(cellValues, SCORE_INDEX));
+        addSimpleCustomField(ticket, LINK_TO_IQB_FIELD_ID, getCellValue(cellValues, LINK_TO_IQB_INDEX));
+        return ticket;
+    }
+
+    private String getCellValue(final List<CellValue> cellValues, final int index) {
+        if (index < cellValues.size()) {
+            return cellValues.get(index).getValue();
+        } else {
+            return org.apache.commons.lang.StringUtils.EMPTY;
```
[the remark this answers is not in the document]
Comment (umutozk) [reply src/main/java/com/devfactory/processautomation/service/process/TicketServiceImpl.java]: There is already a StringUtils (com.devfactory.processautomation.util.StringUtils) imported so this solution is not applicable.

**Pull request merged**